### PR TITLE
fix a heap corruption bug in ems realloc

### PR DESCRIPTION
**Pull request by yamt**

### CODE REVIEW

**Review by wenyongh — COMMENTED** [review]
LGTM

File: samples/mem_allocator/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+cmake_minimum_required(VERSION 3.0)
```
Comment (wenyongh) [inline samples/mem_allocator/CMakeLists.txt:1]: Had better add license header.

File: samples/mem_allocator/main.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
+#include <stdio.h>
```
Comment (wenyongh) [inline samples/mem_allocator/main.c:1]: Should add license header

File: samples/mem_allocator/CMakeLists.txt (hunk 1 above)
Comment (yamt) [reply samples/mem_allocator/CMakeLists.txt:1]: ok. i will add.

Comment (yamt) [reply samples/mem_allocator/CMakeLists.txt:4]: done

File: samples/mem_allocator/main.c (hunk 2 above)
Comment (yamt) [reply samples/mem_allocator/main.c:6]: done

**Pull request merged**